lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -573,16 +748,38 @@ void ScienceSensorsSystem::PostUpdate(const ignition::gazebo::UpdateInfo &_info,
       {
         // Increment for next point in time
         this->dataPtr->timeIdx++;
+
+        // Publish science data at the next timestamp
         this->dataPtr->PublishData();
       }
     }
 
+    // Publish every n iters so that VisualizePointCloud plugin gets it.
+    // Otherwise the initial publication in Configure() is not enough.
+    if (this->dataPtr->repeatPubTimes % 10000 == 0)
+    {
+      this->dataPtr->PublishData();
+      // Reset
+      this->dataPtr->repeatPubTimes = 1;
+    }
+    else
+    {
+      this->dataPtr->repeatPubTimes++;
+    }
+
     // For each sensor, get its pose, search in the octree for the closest
     // neighbors, and interpolate to get approximate data at this sensor pose.
     for (auto &[entity, sensor] : this->entitySensorMap)
     {
       // Sensor pose in lat/lon, used to search for data by spatial coordinates
+      // TODO convert to Cartesian
       auto sensorLatLon = ignition::gazebo::sphericalCoordinates(entity, _ecm);
+      /*
+      // TODO DEBUG what is the sensor attached to?? World? Not robot?
```

Comment:
Are you getting weird values here? Can these debug statements be removed?

Reply:
This block is removed in #83, where the sensor position is converted to Cartesian and tested. In this PR, the sensor position couldn't be tested until we have #83. I'll leave it because otherwise this PR would be going forward in time.